File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {

        int t = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                if (i != t) {
                    nums[t] = nums[i];
                    nums[i] = 0;
                }
                t++;
            }
        }
        
    }
}

File: 283-move-zeroes/move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        slow = fast = 0
        n = len(nums)
        count = 0
        for num in nums:
            if num != 0 : count += 1
        

        while(slow<count and fast<n):
            if(nums[slow] == 0):
                while(nums[fast] == 0):
                    fast += 1
                nums[slow], nums[fast] = nums[fast], 0
                slow += 1
            else:
                slow += 1
                fast += 1
        return nums
            
   